nmostesting/suites/IS1101Test.py

```diff
@@ -0,0 +1,3813 @@
+                            response
+                        )
+                    )
+                if response.status_code != 200:
+                    return test.FAIL(
+                        "The streamcompatibility request for sender {} status has failed: {}".format(
+                            sender_id, response
+                        )
+                    )
+
+                try:
+                    flow_id = response.json()["flow_id"]
+                except json.JSONDecodeError:
+                    return test.FAIL("Non-JSON response returned from Node API")
+                except KeyError as e:
+                    return test.FAIL("Unable to find expected key: {}".format(e))
+
+                if flow_id is None:
+                    return test.FAIL("flow_id is null")
+                valid, response = TestHelper.do_request(
+                    "GET", self.node_url + "flows/" + flow_id
+                )
+                if not valid:
+                    return test.FAIL(
+                        "Unexpected response from the streamcompatibility API: {}".format(
+                            response
+                        )
+                    )
+                if response.status_code != 200:
+                    return test.FAIL(
+                        "The streamcompatibility request for sender {} status has failed: {}".format(
+                            sender_id, response
+                        )
+                    )
+                try:
+                    grain_rate = response.json()["grain_rate"]
+                except json.JSONDecodeError:
+                    return test.FAIL("Non-JSON response returned from Node API")
+                except KeyError as e:
+                    return test.FAIL("Unable to find expected key: {}".format(e))
+
+                if grain_rate != self.get_another_grain_rate(
+                    self.flow_grain_rate[sender_id]
+                ):
+                    return test.FAIL(
+                        "The flow_grain_rate does not match the constraint"
+                    )
+                valid, response = TestHelper.do_request(
+                    "DELETE",
+                    self.compat_url + "senders/" + sender_id + "/constraints/active/",
+                )
+                if not valid:
+                    return test.FAIL(
+                        "Unexpected response from the streamcompatibility API: {}".format(
+                            response
+                        )
+                    )
+                if response.status_code != 200:
+                    return test.FAIL(
+                        "The streamcompatibility request for sender {} status has failed: {}".format(
+                            sender_id, response
+                        )
+                    )
+            return test.PASS()
+        return test.UNCLEAR("No resources found to perform this test.")
+
+    def test_02_03_05_02(self, test):
```

Comment:
The test numbering seems very convoluted.  AFAIK test names should be of the form `test_XX`.  Where they are named `test_XX_XX` this is where additional tests have been added at a later date...

Reply:
A compromise that takes into account the existing sectioning: `test_XX: Senders ...` naming (turning subsection numbers into word prefixes).